specs/time_series_api.adoc

```diff
@@ -0,0 +1,287 @@
+Following are its sub-components based on the functionalities:
+
+* Api Layer : This layer will serve the following responsibilities.
+  ** Serve the time series data stored in time series database.
+  ** The apis internally use a plugin based approach for querying the
+     time-series performance data which eases the task of supporting multiple
+     time-series databases.
+* Aggregator : This module aggregates atomic stats(ex: cluster utilization) to
+  composite(system utilization) stats which will be required by UI for the
+  dashboards.
+
+== Problem description
+
+Tendrl/tendrl-api(a part of tendrl-core) is agnostic of time-series db(a part
+of monitoring stack of tendrl) and its specifics but needs to be able to expose
+data in the time series db if the monitoring stack is installed.
+
+== Use Cases
+
+* If monitoring stack is installed by user, the user should be able to seamlessly
+  access tendrl maintained time series performance data over specified
+  time-ranges.
+
+== Proposed change
+
+* Tendrl/performance-monitoring will expose apis to access performance data in
+  time-series db. This is achieved by:
+  ** Interface to time-series db is designed in the form of plugins specific to
+     a time-series db.
+    *** This means that to support a new time-series db, all it takes is the
+        addition of its apecic interface as a plugin and doesn't require
+        touching any other part of the code.
+    *** The way the plugins work is :
+      **** The interfaces to time-series db are added under the directory
+           tendrl/performance-monitoring/time_series_db/dbplugins of
+           tendrl/performance-monitoring.
+      **** These plugins inherit from a common interface and hence all of them
+           follow a similar basic skeletal structure.
+      **** The plugin framework uses python's meta-class approach and they get
+           registered with the framework when everything in the plugin path is
+           imported but only the plugin configured in trendrl's conf will be
+           used to interface to the time-series db. The way this works is as
+           follows:
+            ***** A singleton manager looks for all/any plugins in directory
+                  tendrl/performance-monitoring/time_series_db/dbplugins and
+                  imports them all.
+            ***** The moment they are imported the meta-class registers an
+                  instance of the plugin with itself.
+            ***** The singleton manager then stores the instance of required
+                  time-series db plugin which will be henceforth used for all
+                  interactions with the time-series db.
+            ***** Since the manager is a singleton, there is only one instance
+                  of it throughout the life of Tendrl/performance-monitoring
+                  application which means all this house-keeping is done only
+                  once during the lifecycle of the application.
```

Comment:
This spec talks about apis that talk to time series data. So I added more details like how the apis actually fetch the details i.e, the api talks to a manager that takes the responsibility of finding the correct plugins that actually talks to time-series db